hazelcast/src/test/java/com/hazelcast/internal/dynamicconfig/AbstractDynamicConfigGeneratorTest.java

```diff
@@ -915,6 +915,15 @@ private void testMap(MapStoreConfig mapStoreConfig) {
         assertEquals(attrConfig.getName(), decAttrConfig.getName());
         assertEquals(attrConfig.getExtractorClassName(), decAttrConfig.getExtractorClassName());
         ConfigCompatibilityChecker.checkMapConfig(expectedConfig, actualConfig);
+
+        assertMapStoresAreEqual(expectedConfig.getMapStoreConfig(), actualConfig.getMapStoreConfig());
```

Comment:
Shouldn't this rather be part of the `com.hazelcast.config.ConfigCompatibilityChecker.MapConfigChecker.isCompatible(MapStoreConfig, MapStoreConfig)`?

Reply:
I am not sure, I thought the reason we use that method is map configs can be different across the cluster as long as compatible (This is not possible for dynamic configs but it's for static configs I believe). And maybe these can be different between map stores hence, not included.

I also tried only using equals, but in that case class equalities create a problem in tests. So we need the logic in `com.hazelcast.config.ConfigCompatibilityChecker.MapConfigChecker.isCompatible` for checking equality of classes but I don't know/think that method needs/can have excluded assertions.

Reply:
https://github.com/hazelcast/hazelcast/pull/20743/commits/737da411c3009049b39e8cc0b598d7869dc0130d